packages/core/src/listeners.ts

```diff
@@ -85,3 +87,39 @@ startCoreListening({
     listenerApi.dispatch(fetchCohortCaseCounts(action.meta.arg?.cohortId));
   },
 });
+
+startCoreListening({
+  matcher: isAnyOf(
+    cohortApiSlice.endpoints.getCohortsByContextId.matchFulfilled,
+    cohortApiSlice.endpoints.getCohortById.matchFulfilled,
+    cohortApiSlice.endpoints.addCohort.matchFulfilled,
+    cohortApiSlice.endpoints.updateCohort.matchFulfilled,
+    cohortApiSlice.endpoints.deleteCohort.matchFulfilled,
+  ),
+  effect: async () => {
+    // Store context id cookie in local storage to make it more resilient to deletion
+    const contextId = Cookies.get("gdc_context_id");
+    if (contextId) {
+      localStorage.setItem("gdc_context_id", contextId);
+    }
+  },
+});
+
+startCoreListening({
+  matcher: isAnyOf(
+    cohortApiSlice.endpoints.getCohortsByContextId.matchPending,
+    cohortApiSlice.endpoints.getCohortById.matchPending,
+    cohortApiSlice.endpoints.addCohort.matchPending,
+    cohortApiSlice.endpoints.updateCohort.matchPending,
+    cohortApiSlice.endpoints.deleteCohort.matchPending,
+  ),
+  effect: async () => {
+    // If cookie has been deleted, restore it from local storage
+    if (!Cookies.get("gdc_context_id")) {
+      const contextId = localStorage.getItem("gdc_context_id");
```

Comment:
Will there be a case when both the localStorage and Cookies wont have this? If so what happens.

Reply:
I believe the backend will create a context id and attach the set-cookie header if the request doesn't have a context id.